src/server_state.cpp

```diff
@@ -764,9 +764,15 @@ void wsrep::server_state::on_primary_view(
     }
     else
     {
-        if (state_ == s_joiner)
+        // There are 2 possible states in which we can be here
+        // 1. s_joined when we have already done initialization
+        // 2. s_initialized when initialization hasn't been done
```

Comment:
The state should end up to `s_initialized` after the server side initialization has been done and `initialized()` is called, so this comment does not make sense.

Reply:
When condition sst_before_init() is true we will also transition from s_initialized to s_joined or s_synced.